lib/cli/Cargo.toml

```diff
@@ -154,6 +155,7 @@ singlepass = [
 cranelift = [
     "wasmer-compiler-cranelift",
     "compiler",
+    "wasmer-wasi/compiler-cranelift"
```

Comment:
I guess this `wasmer-wasi/compiler-cranelift` should disapear. There is no reasn to have a compiler dependancy on Wasi

Reply:
The problem is this: 

There is a "virtual" `wasmer run` command that is available inside of WASIX instances.

For this reason there is a new `ModuleCache` type, which can retrieve WebC images from a registry and compile them into a module so it can be executed as a "WasiProcess". 

This also needs to be available in the CLI so you can do the same thing when (for example) running bash or dash.

I agree that this should be refactored, and I wanted to so so for months.
This can be handled by something like a 

```rust
trait ModuleBuilder {
  fn compile(..) ...
}
```

That way we can drop the compiler features.

I would also argo that `ModuleCache`, `BinPackage`, `BinFactory`, `CmdWasmer` (`lib/wasi/os/command/builtins/cmd_wasmer`) etc don't actually belong in the wasi crate, but in a separate utility crate , which can be used by the CLI and  by Deploy.

Just didn't have any cycles to do that yet.

Reply:
Note also the `Console` abstraction, which runs a given WebC package with a TTY.

Same thing applies to that.

Reply:
Compiler specific features have been removed.